first_order_model/shrink_util.py

```diff
@@ -132,55 +132,49 @@ def set_module(mod, state_dict):
     for key1 in state_dict.keys():
         if key1 not in ['generator', 'kp_detector']:
             continue
+        if force_model is not None and key1 != force_model:
+            continue
         for key, dict_param in state_dict[key1].items():
             if key1 == 'kp_detector':
                 submod_names = ['kp_extractor'] + key.split(".")
             else:
                 submod_names = [key1] + key.split(".")
+            if force_model != None:
+                submod_names = key.split(".")
             #curr_param = get_attr(mod, submod_names)
             # Here you can either replace the existing one
             set_attr(mod, submod_names, dict_param)
             group_name = submod_names[:-1] + ['groups']
             og_groups = get_attr_default(mod, group_name, 1)
             if og_groups != 1:
                 get_attr(mod, group_name[:-1]).groups = dict_param.shape[0]
+                try:
```

Comment:
I really dislike try catch, especially here where you're not printing a useful error. It becomes really hard to debug later

Reply:
I agree, although I'm not sure how to fix this.

Reply:
Just don't use a try catch?